src/Toml/Parser/Value.hs

```diff
@@ -170,10 +171,23 @@ doubleP :: Parser Double
 doubleP = lexeme (signed sc (num <|> inf <|> nan)) <?> "double"
   where
     num, inf, nan :: Parser Double
-    num = float
+    num = floatP
     inf = 1 / 0 <$ string "inf"
     nan = 0 / 0 <$ string "nan"
 
+floatP :: Parser Double
+floatP = check =<< readMaybe . concat <$> sequence [ digits, expo <|> dot ]
```

Comment:
I think that this parser should go `Toml.Parser.Core`. `Toml.Parser.Value` is a very big module, so let's try to not make it bigger 🙂

Reply:
I tried, but it was not a good fit. I had to add so many import to `Toml.Parser.Core` and it didn't fit with any of the functions there... Maybe a refactor of larger scale might work, moving all the functions that are not exported by `Toml.Parser.Value` to `Core`.